tests/test_markers.py

```diff
@@ -388,12 +387,38 @@ def test_extra_str_normalization(self):
         assert str(Marker(rhs)) == f'extra == "{normalized_name}"'
 
     def test_python_full_version_untagged_user_provided(self):
-        """A user-provided python_full_version ending with a + fails to parse."""
-        with pytest.raises(InvalidVersion):
-            Marker("python_full_version < '3.12'").evaluate(
-                {"python_full_version": "3.11.1+"}
-            )
+        """A user-provided python_full_version ending with a + uses Python behaviour."""
+        env = {"python_full_version": "3.11.1+"}
+        assert Marker("python_full_version < '3.12'").evaluate(env)
 
     def test_python_full_version_untagged(self):
         with mock.patch("platform.python_version", return_value="3.11.1+"):
             assert Marker("python_full_version < '3.12'").evaluate()
+
+    @pytest.mark.parametrize(
+        ("marker_string", "environment", "expected"),
+        [
+            ("platform_release >= '20.0'", {"platform_release": "21-foobar"}, True),
+            ("platform_release >= '8'", {"platform_release": "6.7.0-gentoo"}, False),
+            ("platform_version == '27'", {"platform_version": "weird string"}, False),
```

Comment:
This is very much a personal opinion, so feel free to ignore it, but it might be nice to have at least one test case that shows string comparison giving the opposite result from how a human would do a version comparison. Like the one you put in the PR description:

```
("platform_release >= '20'", {"platform_release": "6.7.0-gentoo"}, True)
```

I'm imagining a different algorithm that extracts the longest prefix of each string which constitutes a valid PEP 440 version number and then compares based on that, which I think would pass all the existing test cases, but of course that is not the algorithm PEP 508 specifies so IMO it'd be nice to have a test case that would catch that.

Reply:
I agree -- this would be good to add here.

Reply:
I did do that in [`2ce09d9` (#816)](https://github.com/pypa/packaging/pull/816/commits/2ce09d963b6b9b954dfe1a09639e7878acb9925f). I forgot to mention that, sorry! :sweat_smile: